### DISCUSSION TITLE
Interesting video about using matter on esp32

### DISCUSSION THREAD

**Discussion by SnowballsChance** [HomeSpan/HomeSpan#330] [General]
Just saw [this](https://youtu.be/eW1sPzDY3S0) and thought it was an interesting initial project for making esp32 matter devices.
Thought the community here might find it interesting. 
Cheers!

**Comment by HomeSpan** [comment]
Thanks for posting this link - definitely very interesting.  I've been waiting for Apple to release a Matter-integrated version of the Home App to try out the new protocol.  Though Apple has said they will continue to support the HAP HomeKit protocol, if Matter really takes off, my guess is it will surpass the functionality of HomeKit, making libraries like HomeSpan less useful.  To the extent that Matter is a truly open, updated, and fully documented protocol (it is supposed to be), and Apple does not create any proprietary overlays that require special licensing, the move to Matter will be a welcome change (even if that means HomeSpan is no longer needed 😮).

**Comment by danilkorotkov** [comment]
Matter SDK for ESP32 is ready but not for ARDUINO
https://github.com/espressif/esp-matter

**Reply by danilkorotkov** [reply]
esp sdk has a different device initialization(main.cpp ) than the original CHIP project(ZAP-generated)

**Reply by danilkorotkov** [reply]
compiled and flashed 4 examples but home app can't connect to this accessory
